### ISSUE TITLE
tox reports 2 failed tests for py37

### ISSUE THREAD

**Issue by cshields143** [monocongo/climate_indices#419]
**Describe the bug**
`test_compute.py::test_transform_fitted_pearson` throws an AssertionError:

```
E       AssertionError: 
E       Not equal to tolerance rtol=1e-07, atol=0.001
E       Transformed Pearson Type III fitted values not computed as expected
E       x and y -inf location mismatch:
E        x: array([[ 0.207576,  0.891472, -1.00923 , ...,  0.447044,  0.735171,
E               -0.072223],
E              [ 0.724334,  0.393734, -0.988155, ..., -0.266911,  0.684934,...
E        y: array([[ 0.207576,  0.891472, -1.00923 , ...,  0.447044,  0.735171,
E               -0.072223],
E              [ 0.724334,  0.393734, -0.988155, ..., -0.266911,  0.684934,...
```

Ditto for `test_indices.py::test_spi`:

```
E       AssertionError: 
E       Not equal to tolerance rtol=1e-07, atol=0.01
E       SPI/Pearson values for 6-month scale not computed as expected
E       Mismatched elements: 120 / 1476 (8.13%)
E       Max absolute difference: 6.18
E       Max relative difference: 2.
E        x: array([nan, nan, nan, ..., nan, nan, nan])
E        y: array([nan, nan, nan, ..., nan, nan, nan])
```

**To Reproduce**
1. `git clone https://github.com/monocongo/climate_indices`
2. `cd climate_indices`
3. `conda create -n python37 python=3.7`
4. `conda activate python37`
5. `pip install -r requirements.txt`
6. `pip install tox`
7. `tox -e py37`

**Expected behavior**
`py37: commands succeeded`
`congratulations :)`

**Desktop (please complete the following information):**
- OS: Ubuntu
- Version: 20.04.2 LTS
- Architecture: 64bit

**Additional context**
All tests *do* pass with python 3.6

**Comment by cshields143** [comment]
Obviously I have the source, so I can do some debugging and see if I can't figure out why.

Maybe I don't have the environment or testing pipeline set up right?

**Comment by cshields143** [comment]
Some "progress" has been made on the first failed test: `scipy.stats.pearson3.cdf` returns different values when run on different python versions. For more info: https://github.com/cshields143/climate_indices_issue_419a

It returns `0` in later versions (3.7, 3.8) when it returns `nan` in earlier versions (3.6)

**Comment by monocongo** [comment]
Chris this is a very interesting discovery. I'm not sure what to make of it
yet. We might want to take it up with scipy folks? In any event thanks for
your detective work on this so far!

—James

**Comment by cshields143** [comment]
`scipy.stats.pearson3` is computing the `cdf` on a column-by-column basis. We see that the values in the original data which are causing our "invalid 0s" are, all of them, the smallest values in those columns--all at least a full standard deviation away from each column's mean. Given how CDFs work, these *should* give us 0s; it's actually the earlier version, which returns `nan` instead of `0`, that is mistaken

Why are we getting different answers across different Python versions? Because based on how I've described the "steps to reproduce" above, `tox` will install the latest version available of each dependency (presumably it uses the list of deps in `setup.py` instead of `requirements.txt`--the former does not specify explicit versions, while the latter does). When we execute `tox` in Python 3.6, the latest version of SciPy it can get is 1.5.4; with Python 3.7, it is able to use 1.6.1

This can be confirmed by closely looking at the first few lines of `tox -e py36`...

    .package create: /home/chris/Projects/climate_indices/master/.tox/.package
    .package installdeps: setuptools>=40.8.0, wheel
    py36 create: /home/chris/Projects/climate_indices/master/.tox/py36
    py36 installdeps: codecov, coverage, pytest
    py36 inst: /home/chris/Projects/climate_indices/master/.tox/.tmp/package/1/climate_indices-1.0.9.tar.gz
    py36 installed: attrs==20.3.0,...scipy==1.5.4...

...and the first few lines of `tox -e py37`:

    .package recreate: /home/chris/Projects/climate_indices/master/.tox/.package
    .package installdeps: setuptools>=40.8.0, wheel
    py37 recreate: /home/chris/Projects/climate_indices/master/.tox/py37
    py37 installdeps: codecov, coverage, pytest
    py37 inst: /home/chris/Projects/climate_indices/master/.tox/.tmp/package/1/climate_indices-1.0.9.tar.gz
    py37 installed: attrs==20.3.0,...scipy==1.6.1...

SciPy 1.6+ does not support Python 3.6 (only 3.7+), hence why 1.5.4 is used instead. It turns out the `pearson3` module was *buggy* prior to 1.6 (see https://github.com/scipy/scipy/pull/12640), hence the different answers in different versions.

Are all of the tests passing in CircleCI? If at some point in the actual testing pipeline the explicit requirement of SciPy 1.4.1 is used instead of "whatever the latest available is", this means the old, buggy behavior would be present in *any* Python version--hence why this wouldn't have come up before.

**How to fix**:

- Drop support for Python 3.6; update the various config files that mention it
- Explicitly require SciPy 1.6+
- Update the `pearson3_monthly.npy` file in `tests/fixture` to accurately reflect non-buggy behavior

If all of this looks good (I don't know how necessary 3.6 is for the project), I don't mind throwing this together into a pull request

**Comment by monocongo** [comment]
@cshields143 if you still have an interest in this and can make the proposed changes then it will be quite helpful in order to get the tests back in shape. Your in-depth analysis and proposed solution look good, it seems that you've cracked this nut.

**Comment by monocongo** [comment]
Also no, there isn't a requirement for 3.6 support, though it's a goal to make the package as widely useful as possible so if we can keep 3.6 support somehow with some dependency management/requirements.txt etc. tricks then that's even better, maybe best left for later separate from this effort, the main goal is to get the tests stood up again since that's the main stink in this project for now.

**Comment by WeatherGod** [comment]
Additional discussion and debugging of both test failures can be found at https://github.com/monocongo/climate_indices/pull/450

**Comment by monocongo** [comment]
Fixed by #450 

Thanks again for your contributions @cshields143 @WeatherGod
